src/agent/src/sandbox.rs

```diff
@@ -151,7 +151,11 @@ impl Sandbox {
     pub fn remove_sandbox_storage(&self, path: &str) -> Result<()> {
         let mounts = vec![path.to_string()];
         remove_mounts(&mounts)?;
-        fs::remove_dir_all(path).context(format!("failed to remove dir {:?}", path))?;
+        // "remove_dir_all" can fail if it's a read-only filesystem.
+        // We don't want to fail the container deletion because of that.
+        if let Err(err) = fs::remove_dir_all(path) {
```

Comment:
are there scenarios where we *don't* want to remove_dir_all?

ie, what if we have a bind mount that overwrites an existing path?
vs, what if we had to create /configs/ and /configs/path to mount at /configs/path ?

Alternatively, are there any scenarios where we'd be calling remove_sandbox_storage outside of Container removal? While we should cleanup after ourselves, we should probably either fully track/do it right (cleanup if we had to make the path to do the mount), or skip.

I'm curious how this is managed in runc/libcontainer.

Reply:
While at it, could you change it to `remove_dir`? `remove_dir_all` is too dangerous and we've seen terrible bugs with such API. Here we just want to remove the mount point path itself. If there is anything inside it, it is not a mount point path created by the agent. And we should not remove it in such a case.